spatialprofilingtoolbox/ondemand/providers/squidpy_provider.py

```diff
@@ -65,13 +68,16 @@ def _perform_computation(
     @staticmethod
     def _form_cells_dataframe(arrays: CellDataArrays) -> DataFrame:
         features = tuple(f.symbol for f in arrays.feature_names.names)
-        rows = []
-        zipped = zip(arrays.identifiers, arrays.phenotype, arrays.location.transpose())
-        for identifier, phenotype, location in zipped:
-            location_list = [location[0], location[1]]
-            vector = list(OnDemandProvider.extract_binary(phenotype, len(features)))
-            row = tuple(vector + [identifier] + location_list)
-            rows.append(row)
+
+        phenotype_uint8 = arrays.phenotype.view(np_uint8) \
+            .reshape(arrays.phenotype.shape[0], 8)
+        binary = unpackbits(phenotype_uint8, axis=1)
+        phenotypes = binary[:, -len(features):][:, ::-1].astype(int)
+
+        identifiers = arrays.identifiers.reshape(-1, 1)
+        location = arrays.location.transpose()
+        rows = hstack((phenotypes, identifiers, location))
+
```

Comment:
I think I found the source of the failing squidpy tests. These lines always seem to create 0 values for the phenotype membership flags. Using debug statements I found that the previous `rows` and `df` looked like:

```
(1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 300, 4829, 1601)
(1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 301, 5198, 1612)
(1, 1, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 302, 5004, 1614)
...

                           B2M  B7H3  CD68  CD8  ...  CD4  CD56  pixel x  pixel y
histological_structure_id                        ...                             
300                          1     1     0    0  ...    1     0     4829     1601
301                          1     1     0    0  ...    0     0     5198     1612...
```
and the current:
```
[   0.    0.    0.    0.    0.    0.    0.    0.    0.    0.    0.    0.
    0.    0.    0.    0.    0.    0.    0.    0.    0.    0.    0.    0.
    0.    0.  300. 4829. 1601.]
[   0.    0.    0.    0.    0.    0.    0.    0.    0.    0.    0.    0.
     0.    0.    0.    0.    0.    0.    0.    0.    0.    0.    0.    0.
     0.    0.  301. 5198. 1612.]
...
                           B2M  B7H3  CD68  CD8  ...  CD4  CD56  pixel x  pixel y
histological_structure_id                        ...                             
300.0                      0.0   0.0   0.0  0.0  ...  0.0   0.0   4829.0   1601.0
301.0                      0.0   0.0   0.0  0.0  ...  0.0   0.0   5198.0   1612.0

Reply:
I just reverted this to pass the tests.